components/uniset/src/conversions.rs

```diff
@@ -0,0 +1,123 @@
+use crate::utils::deconstruct_range;
+use crate::UnicodeSet;
+use std::{
+    convert::TryFrom,
+    ops::{Range, RangeBounds, RangeFrom, RangeFull, RangeInclusive, RangeTo, RangeToInclusive},
+};
+
+fn try_from_range_impl(range: impl RangeBounds<char>) -> Result<UnicodeSet, (u32, u32)> {
+    let (from, till) = deconstruct_range(range);
+    if from < till {
+        let set = vec![from, till];
+        Ok(UnicodeSet::try_from(set).unwrap())
+    } else {
+        Err((from, till))
```

Comment:
Please make an `Error` enum in `lib.rs` and return an item from that enum.  For example:

```rust
// lib.rs
#[derive(Debug, PartialEq)]
enum Error {
    InvalidRange(u32, u32),
}

// when you encounter the error
Err(Error::InvalidRange(from, till))
```

Reply:
agreed and changed, with the addition of `InvalidSet(Vec<u32>)` for the `try_from::<Vec<u32>>` conversion